src/Storage/TSStorageManager.m

```diff
@@ -238,24 +238,40 @@ - (BOOL)databasePasswordAccessible
 - (NSData *)databasePassword
 {
     [SAMKeychain setAccessibilityType:kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly];
-    NSString *dbPassword = [SAMKeychain passwordForService:keychainService account:keychainDBPassAccount];
-
-    if (!dbPassword) {
-        dbPassword = [[Randomness generateRandomBytes:30] base64EncodedString];
-        NSError *error;
-        [SAMKeychain setPassword:dbPassword forService:keychainService account:keychainDBPassAccount error:&error];
-        if (error) {
-            // Sync log to ensure it logs before exiting
-            NSLog(@"Exiting because we failed to set new DB password. error: %@", error);
-            exit(1);
-        } else {
-            DDLogError(@"Succesfully set new DB password. First launch?");
+
+    NSError *keyFetchError;
+    NSString *dbPassword =
+        [SAMKeychain passwordForService:keychainService account:keychainDBPassAccount error:&keyFetchError];
+
+    if (keyFetchError) {
+        switch (keyFetchError.code) {
+            case errSecItemNotFound:
+                dbPassword = [self createAndSetNewDatabasePassword];
+                break;
+            default:
+                [NSException raise:@"Serious error when getting DB password from keychain."
```

Comment:
Can you us a constant for the exception name, like `NSString *const TSStorageManagerInaccessiblePaswordException = @"TSStorageManagerInaccessiblePaswordException"`

and similarly in `createAndSetNewDatabasePassword`. That paves the way for any future handling based on name.